_posts/2024-02-21-securing-wildfly-apps-okta-openshift.adoc

```diff
@@ -0,0 +1,260 @@
+* Email: *alice@example.org*
+* User Name: *alice@example.org* 
+* From the dropdown list for _Password_, choose _Set by admin_ and set a password that fits the password requirements. 
+* Uncheck _User must change password on first login_ and hit `Save`. For more information about how to add a user manually https://help.okta.com/en-us/content/topics/users-groups-profiles/usgp-add-users.htm[click here].
+. Click on the _Applications_ tab on the left hand side. Navigate to _Applications_ and click on _simple-webapp-okta_. Under the _Assignments_ tab, click on *Assign*, then click on *Assign to People* and select *Alice*. For more information about how to assign a user to an application, please see https://support.okta.com/help/s/article/How-To-Assign-An-User-To-An-Application?language=en_US[here].
+
+=== Advanced OpenID Configuration
+In order to extract the user information through the id token, we will need to configure a custom authorization server. Follow the steps below for this:
+
+. Under the *Security* dropdown, select *API* and click on the edit button next to the *Default* server. 
+. Under the *Claims* tab, click on the *Add Claim* button and to add a claim with the following configurations: 
+* Name: *IDTClaim*
+* Include in token type: *ID Token* and keep it at always. 
+* Value type: *Expression*
+* Value: *appuser.userName* and hit `Save`. 
+You can learn more about Expression Language Reference for Okta https://developer.okta.com/reference/okta_expression_language/[here].
+. Now go back to the _API_ menu and copy the *Issuer URI* for the _Default_ server. 
+. This will be used as the provider url when configuring our WildFly application.
+
+== Add Helm Configuration
+. Keep the URL obtained from the last step. 
+. Switch to the `charts` directory in the `simple-webapp-okta` example.
+```
+cd /PATH/TO/ELYTRON/EXAMPLES/simple-webapp-okta/charts
+```
+Notice there’s a helm.yaml file in this directory with the following content:
+```
+build:
+  uri: https://github.com/wildfly-security-incubator/elytron-examples.git
+  contextDir: simple-webapp-okta
+deploy:
+  replicas: 1
+  env:
+    - name: OIDC_PROVIDER_URL
+      value: <okta_URL>         <1>
+    - name: OIDC_CLIENT_ID
+      value: <client_id>        <2>
+    - name: OIDC_CLIENT_SECRET
+      value: <client_secret>    <3>
+```
+We will need to make the following changes locally to the helm chart: 
+
+<1> Replace *OKTA_URL* with the *provider url* you copied in the last step.
+<2> Replace *client_id* with the *Client ID* listed on the Okta console. Navigate to the _General_ tab for our application on the Okta admin console and copy the _Client Id_ and add it beside the `client-id` attribute. 
+<3> Replace *client_secret* with your *Client Secret* for this application listed on Okta. 
+
+Save this file and close it.
+
+== Configure the Deployment Settings
+We can use the resources under 'elytron-oidc-client' to configure the WildFly server. We can do this in two different ways: 
```

Comment:
s/We can use the resources under 'elytron-oidc-client' to configure the WildFly server. We can do this in two different ways/Applications deployed to WildFly can be secured with OIDC in a couple different ways:

Reply:
fixed!